File: trunk/dotNet/TestRunner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace DotNetPerformanceTest
{
    class TestRunner
    {
        private List<SomeTest> _tests = new List<SomeTest>();
        
        public TestRunner()
        {
            _tests.Add(new IntRandomTest());
            _tests.Add(new DoubleRandomTest());
            _tests.Add(new DynamicArrayAccessTimeTest());
            _tests.Add(new DynamicArrayInsertRemoveTest());
            _tests.Add(new LinkedListAccessTimeTest());
            _tests.Add(new LinkedListInsertRemoveTest());
        }
        
        public void Do()
        {
            foreach (SomeTest test in _tests)
            {
                Console.WriteLine("running test: " + test.Name);
                test.Do();
            }
        }

        public void WriteResult()
        {
            Console.WriteLine();
            foreach (SomeTest test in _tests)
            {
                Console.WriteLine(test.Time + "\t ms. " + test.Name);
            }
        }
    }
}

File: Java/src/JavaPerformanceTest/TestRunner.java
package JavaPerformanceTest;

import java.util.ArrayList;

public class TestRunner
{
    private ArrayList<SomeTest> _tests = new ArrayList<SomeTest>();
    
    public TestRunner()
    {
        _tests.add(new IntRandomTest());
        _tests.add(new DoubleRandomTest());
        _tests.add(new DynamicArrayAccessTimeTest());
        _tests.add(new DynamicArrayInsertRemoveTest());
        _tests.add(new LinkedListAccessTimeTest());
        _tests.add(new LinkedListInsertRemoveTest());
    }
    
    public void Do()
    {        
        for (int i = 0; i < _tests.size(); ++i)
        {
            SomeTest test = _tests.get(i);
        	System.out.println("running test: " + test.getName());
            test.Do();
        }
    }

    public void WriteResult()
    {
    	System.out.println();
    	for (int i = 0; i < _tests.size(); ++i)
        {
            SomeTest test = _tests.get(i);
        	System.out.println(test.getTime() + "\t ms. " + test.getName());
        }
    }
}